src/xmipp/libraries/reconstruction/angular_distance.cpp

```diff
@@ -573,3 +625,114 @@ void ProgAngularDistance::computeWeights()
     DF2weighted.removeDisabled();
     DF2weighted.write(fn_out+"_weights.xmd");
 }
+
+void ProgAngularDistance::euler2quat(   double rot, double tilt, double psi,
+                                        double q[4] )
```

Comment:
Use "std::array" or "std::vector" instead of a C-style array.

Reply:
Yes, it is a better idea to use std::array instead of C arrays. However, I am calling these functions with slices of a Matrix2D. This makes it impossible to use std::array (or std::vector) without copying. The best alternative would be to use std::span, but it is a C++20 feature